triggers/runner/src/main/scala/com/digitalasset/daml/lf/engine/trigger/TriggerLogContext.scala

```diff
@@ -132,6 +152,20 @@ object TriggerLogContext {
       loggingContext,
       entries,
       TriggerLogSpan(BackStack(span)),
+      (_, _) => (),
+    ).enrichTriggerContext()(f)
+  }
+
+  private[trigger] def newRootSpanWithCallback[A](
+      span: String,
+      callback: (String, TriggerLogContext) => Unit,
+      entries: (String, LoggingValue)*
+  )(f: TriggerLogContext => A)(implicit loggingContext: LoggingContextOf[Trigger]): A = {
```

Comment:
New factory method for creating root log contexts with the callback parameter specified. I've intentionally made this package private as we generally don't want non-testing code to use this.